src/System Application/App/Telemetry/src/Logging/TelemetryLoggersImpl.Codeunit.al

```diff
@@ -23,24 +25,40 @@ codeunit 8709 "Telemetry Loggers Impl."
 
     procedure Register(TelemetryLogger: Interface "Telemetry Logger"; Publisher: Text)
     begin
+        if RegisteredPublishers.Contains(Publisher) then
+            // Note: this tag won't be available for ISVs as far as I understand
+            Session.LogMessage('0000G7J', StrSubstNo(MultipleTelemetryLoggersFoundErr, Publisher), Verbosity::Warning, DataClassification::SystemMetadata, TelemetryScope::All, 'Category', TelemetryLibraryCategoryTxt)
+        else begin
+            RegisteredTelemetryLoggers.Add(TelemetryLogger);
+            RegisteredPublishers.Add(Publisher);
+        end;
+
         if Publisher = CurrentPublisher then
-            if not IsLoggerFound then begin
-                CurrentTelemetryLogger := TelemetryLogger;
-                IsLoggerFound := true;
-            end else
-                Session.LogMessage('0000G7J', StrSubstNo(MultipleTelemetryLoggersFoundErr, Publisher), Verbosity::Warning, DataClassification::SystemMetadata, TelemetryScope::All, 'Category', TelemetryLibraryCategoryTxt)
+            if not IsLoggerFromCurrentPublisherFound then begin
+                TelemetryLoggerFromCurrentPublisher := TelemetryLogger;
+                IsLoggerFromCurrentPublisherFound := true;
+            end;
```

Comment:
This logic may not be most performant. It will run on very `Register` (could be multiple times per `LogMessage`), but it only save time once per `LogMessage`